packages/cozy-stack-client/src/DocumentCollection.spec.js

```diff
@@ -986,4 +986,41 @@ describe('DocumentCollection', () => {
       })
     })
   })
+
+  describe('handleMissingIndex', () => {
+    const collection = new DocumentCollection('io.cozy.triggers', client)
+    const selector = {
+      'message.account': 'ca7b7f1'
+    }
+    const response = {
+      rows: [
+        {
+          doc: {
+            _id: '123',
+            language: 'query',
+            views: {
+              123456: {
+                map: {
+                  fields: {
+                    'message.account': 'asc'
+                  }
+                }
+              }
+            }
+          }
+        }
+      ],
+      total_rows: 1
+    }
+
+    it('should work with undefined indexedFields', async () => {
+      client.fetchJSON.mockResolvedValue(response)
+      expect(
+        await collection.handleMissingIndex(selector, {
+          indexedFields: undefined,
+          sort: undefined
+        })
+      )
+    })
```

Comment:
Je vois pas comment ton test vient tester l'erreur qu'on a eu précédemment ? 

Il faudrait tester que `getIndexNameFromFields` soit appelé avec l'objet attendu par exemple, non ?

Reply:
Si `getIndexNameFromFields` plante pour x raison (là en l’occurrence c'est le array.join qui plantait puisque array était undefined) ça plante le test.

Reply:
OK mais dans ce cas, ça ne veut pas dire `it should work with undefined indexedFields` mais plus `it should not throw with undefined indexedFields`. 

Aussi c'est pas la peine de mettre une response aussi longue puisque tu testes pas le retour. Un test qui vérifie que rien ne throw serait plus pertinent et plus compréhensible si c'est ce que tu cherches à faire. 

Après perso, je reste convaincu que le bon test c'est de vérifier le comportement ce cette méthode à savoir : 
- que ça doit appeler soit createIndex, soit migrateUnamedIndex avec les bons arguments (dont le indexName qui est récupéré à partir des indexedFields) 

ça permettrait de s'assurer que ça fonctionne dans tous les cas. Car là ça fait quoi si y a pas de sélecteur ou de sort ? (question vraiment naive, j'ai pas checké où on était dans CC et si c'était possible)

Reply:
La response est le minimum vitale pour que ça passe dans toute les méthodes correctement, sinon ça pète d'une manière ou d'une autre.

Sans selector ça devrait passer, les cas sont gérés dans `getIndexFields` et `findExistingIndex` de ce que je vois (j'ai pas testé). Sans sort ça fonctionne puisque c'est le cas testé également (et `getIndexFields` s'en débrouille aussi si je comprends bien le code).

J'avance pas sur mes cartes en cours, dont celle-ci qui traine depuis 10 jours. Je te propose de me le noter pour faire des tests de comportement de manière plus globale, mais au moins le bug est fixé et @poupotte peut avancer sur les tests, voire valider la carte si tout est ok.
Je reviendrai dessus après la carte en cours sur l'ouverture du pdf en natif. Qu'en penses-tu ?

Reply:
Let's merge it. We'll add a few tests with @paultranvan ;)